fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -92,47 +92,6 @@ impl<T> Interpreter<PredicateStorage, T> {
     }
 }
 
-impl<S, Tx> Interpreter<S, Tx>
-where
-    S: InterpreterStorage,
-    Tx: ExecutableTransaction,
-{
-    pub(crate) fn run_call(&mut self) -> Result<ProgramState, RuntimeError> {
-        loop {
-            if self.registers[RegId::PC] >= VM_MAX_RAM {
-                return Err(PanicReason::MemoryOverflow.into());
-            }
-
-            let state = self.execute().map_err(|e| {
-                e.panic_reason()
```

Comment:
Do you not need to carry this mapping over to the refactor?

Reply:
No. The old code was wrapping the panic reason in a RuntimeError and then later undoing that (via implicit `?`). Now that we're combining `run_program` and `run_call`, the error can be propagated directly.